### ISSUE TITLE
SSH connection's aren't closed properly

### ISSUE THREAD

**Issue by mreithub** [ondevice/ondevice#6]
When running `ondevice ssh` and exiting, the `ondevice daemon` doesn't seem to properly close the TCPHandler instance's socket.

This causes troubles with connection limits as well as dangling port forwards.

I've already pushed improvements in terms of event propagation within the application so all the necessary bits and pieces for this to be solved should be there by now.

**Comment by mreithub** [comment]
This only seems to affect the `v0.4.1` release and not the master branch.

**Comment by mreithub** [comment]
I've noticed that this hadn't been fixed (apparently my attempts of testing this were flawed) and was still an issue with v0.4.2.

A relatively simple way to reproduce the issue is (assuming you've got a webserver running on localhost and put a hello.txt in its DocumentRoot):

    $ ondevice ssh $devId -R 1234:localhost:80 curl -s http://localhost:1234/hello.txt
    hello world

This command works the first time, but since the TCPHandler isn't properly notified about the tunnel being closed, it won't terminate its connection to the SSH server (which in turn keeps listening to port 1234 even after the client has exited).

When running the command again, the port is already in use (and worse: listens for connections but doesn't respond - so any request will result in a timeout).

I've finally fixed this issue now in d18be92 (with preparations in ef729cfc).  
And I've published a bugfix release: v0.4.3
